cli/js/files.ts

```diff
@@ -26,71 +26,84 @@ let OP_WRITE = -1;
  *
  *       const file = Deno.openSync("/foo/bar.txt", { read: true, write: true });
  *
- * Requires `allow-read` and `allow-write` permissions depending on mode.
+ * Requires `allow-read` and `allow-write` permissions depending on openMode.
  */
-export function openSync(path: string, mode?: OpenOptions): File;
+export function openSync(path: string, openMode?: OpenOptions): File;
 
 /** Synchronously open a file and return an instance of the `File` object.
  *
  *       const file = Deno.openSync("/foo/bar.txt", "r");
  *
- * Requires `allow-read` and `allow-write` permissions depending on mode.
+ * Requires `allow-read` and `allow-write` permissions depending on openMode.
  */
-export function openSync(path: string, mode?: OpenMode): File;
+export function openSync(
+  path: string,
+  openMode?: OpenMode,
+  mode?: number
+): File;
 
 /**@internal*/
 export function openSync(
   path: string,
-  modeOrOptions: OpenOptions | OpenMode = "r"
+  modeOrOptions: OpenOptions | OpenMode = "r",
+  mode?: number
 ): File {
-  let mode = null;
+  let openMode = null;
   let options = null;
 
   if (typeof modeOrOptions === "string") {
-    mode = modeOrOptions;
+    openMode = modeOrOptions;
   } else {
     checkOpenOptions(modeOrOptions);
     options = modeOrOptions;
+    mode = options.mode;
   }
 
-  const rid = sendSyncJson("op_open", { path, options, mode });
+  const rid = sendSyncJson("op_open", { path, options, openMode, mode });
   return new File(rid);
 }
 
 /** Open a file and resolve to an instance of the `File` object.
  *
  *     const file = await Deno.open("/foo/bar.txt", { read: true, write: true });
  *
- * Requires `allow-read` and `allow-write` permissions depending on mode.
+ * Requires `allow-read` and `allow-write` permissions depending on openMode.
  */
 export async function open(path: string, options?: OpenOptions): Promise<File>;
 
 /** Open a file and resolves to an instance of `Deno.File`.
  *
  *     const file = await Deno.open("/foo/bar.txt, "w+");
  *
- * Requires `allow-read` and `allow-write` permissions depending on mode.
+ * Requires `allow-read` and `allow-write` permissions depending on openMode.
  */
-export async function open(path: string, mode?: OpenMode): Promise<File>;
+export async function open(
+  path: string,
+  openMode?: OpenMode,
+  mode?: number
+): Promise<File>;
```

Comment:
This signature seems not ok to me, also not tested. 
https://deno.land/std/style_guide.md#exported-functions-max-2-args-put-the-rest-into-an-options-object

Reply:
Sure, just let me know whether you prefer (1) this signature to go back to omitting `mode?: number`, or (2) I add a test for this signature.

Reply:
What about `mode: number | OpenMode` ?

Reply:
@dubiousjim omit this signature